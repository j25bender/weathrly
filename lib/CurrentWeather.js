import React from 'react';
import '../styles/CurrentWeather.css';
import PropTypes from "prop-types";

function CurrentWeather(props) {
  //eslint-disable-next-line max-len 
  const { location, icon, condition, day, high, low, tempNum, sentence } = props.forecast;

  return (
    <div className="current">
      <h2 className="currentHeader">CURRENT WEATHER</h2>
      <div className="weather-box" id="current-band">
        <div id="location">{ location }</div>
        <img className="weather-icon" src={ icon } />
        <div id="condition">{ condition }</div>
      </div>
      <div className="weather-box" id="day-hi-lo">
        <div id="day">{ day }</div>
        <div className="hi">Expected High { high }˚F</div>
        <div className="low">Expected Low { low }˚F</div>
      </div>
      <div className="weather-box" id="temp-sentence">
        <div className="current-temp">Current Temperature { tempNum }˚F</div>
        <div id="sentence">{ sentence }</div>
      </div>
    </div>
  );
}

CurrentWeather.propTypes = {
  forecast: PropTypes.object
};

export default CurrentWeather;