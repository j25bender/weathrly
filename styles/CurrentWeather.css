.current {
    color: #fff;
    display: flex;
    justify-content: space-between;
    border: 0.02rem solid ghostwhite;
    padding: 2rem;
    margin: 1rem;
}

.currentHeader {
    color: #fff;
    font-family: "Open Sans";
    font-size: 1.8rem;
    padding: 0 0 1rem 1rem;
    text-align: left;
}
 
.weather-box {
    align-items: center;
    display: flex;
    flex: 1;
    flex-direction: column;
    justify-content: space-between;
}

.weather-icon {
    width: 15rem;
}

@media screen and (max-width: 815px) {
    .current {
        border: none;
        flex-direction: column;
        margin: 0;
        padding: 0;
    }
    
    .weather-box {
        border: 0.02rem solid ghostwhite;
        margin: 1rem;
        padding: 2rem;
    }

    .currentHeader {
        text-align: center;
    }
}