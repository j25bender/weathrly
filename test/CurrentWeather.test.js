import React from 'react';
import { shallow, mount } from 'enzyme';
import CurrentWeather from "../lib/CurrentWeather";
import cleanData from '../lib/cleanData';
import mockData from '../lib/mockData';
import 'jest-localstorage-mock';

describe('CurrentWeather Tests ', () => {
    let filteredData;
    let wrapper;
    
    beforeEach(() => {
        filteredData = cleanData(mockData);
        wrapper = mount(<CurrentWeather forecast={filteredData.CurrentObject}/>);
    });

    it('CurrentWeather component should exist', () => {
        expect(wrapper).toBeDefined()
        expect(wrapper.find('.currentHeader').text()).toEqual('CURRENT WEATHER')        
    });

    it('CurrentWeather should be a function', () => {
        expect(typeof CurrentWeather).toEqual('function');
    });

    it('CurrentWeather should render 1 CurrentWeather component', () => {
        expect(wrapper.find('CurrentWeather').length).toEqual(1);
        expect(CurrentWeather).toMatchSnapshot();
    });
  
    it('CurrentWeather component should render a location, high / low temp, condition, current temp, day, a summary and an icon via an image url', () => {
        expect(wrapper.find('CurrentWeather').props().forecast.location).toEqual('Louisville, KY')

        expect(wrapper.find('CurrentWeather').props().forecast.condition).toEqual('Partly Cloudy')

        expect(wrapper.find('CurrentWeather').props().forecast.high).toEqual('51')

        expect(wrapper.find('CurrentWeather').props().forecast.low).toEqual('32')

        expect(wrapper.find('CurrentWeather').props().forecast.tempNum).toEqual(46)

        expect(wrapper.find('CurrentWeather').props().forecast.sentence).toEqual('Sun and clouds mixed. High 51F. Winds NE at 10 to 15 mph.')
        
        expect(wrapper.find('CurrentWeather').find('.weather-icon').length).toEqual(1);

        expect(wrapper.find('CurrentWeather').find('.weather-box').length).toEqual(3);
        
        expect(wrapper.find('CurrentWeather').props().forecast.icon).toEqual('http://icons.wxug.com/i/c/j/mostlycloudy.gif')
    });
});