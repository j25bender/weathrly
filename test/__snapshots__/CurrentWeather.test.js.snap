// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeather Tests  CurrentWeather should render 1 CurrentWeather component 1`] = `[Function]`;
